management/virtualmachine/client.go

```diff
@@ -118,6 +119,25 @@ func (vm VirtualMachineClient) GetRole(cloudServiceName, deploymentName, roleNam
 	return role, nil
 }
 
+// AddRole adds a Virtual Machine to a deployment of Virtual Machines, where role name = VM name
+// See https://msdn.microsoft.com/en-us/library/azure/jj157186.aspx
+func (vm VirtualMachineClient) AddRole(cloudServiceName string, deploymentName string, role Role) (management.OperationID, error) {
+	if cloudServiceName == "" {
```

Comment:
I feel like this method should get PersistentVMRole as input, not just role. We try to keep inputs of methods and underlying requests matching 1:1 as much as we can. But I might be wrong I'm reading docs on mobile.